src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.Tests/GitHubClientTests.cs

```diff
@@ -0,0 +1,158 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.DarcLib;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Primitives;
+using Moq;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.DotNet.Darc.Tests
+{
+    public class SimpleCacheEntry : ICacheEntry
+    {
+        private object _key;
+        private object _value;
+
+        public SimpleCacheEntry(object key)
+        {
+            _key = key;
+        }
+
+        public object Key => _key;
+
+        public object Value { get => _value; set => _value = value; }
+        public DateTimeOffset? AbsoluteExpiration { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public TimeSpan? AbsoluteExpirationRelativeToNow { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public TimeSpan? SlidingExpiration { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+        public IList<IChangeToken> ExpirationTokens => throw new NotImplementedException();
+
+        public IList<PostEvictionCallbackRegistration> PostEvictionCallbacks => throw new NotImplementedException();
+
+        public CacheItemPriority Priority { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+        public long? Size { get => 0; set => Size = 0; }
+
+        public void Dispose() { }
+    }
+
+    public class SimpleCache : IMemoryCache
+    {
+        public int CacheHits { get; set; }
+        public int CacheMisses { get; set; }
+        public ConcurrentDictionary<object, ICacheEntry> cache = new ConcurrentDictionary<object, ICacheEntry>();
+
+        public ICacheEntry CreateEntry(object key)
+        {
+            var newEntry = new SimpleCacheEntry(key);
+            return cache.AddOrUpdate(key, new SimpleCacheEntry(key), (existingKey, existingValue) => newEntry);
+        }
+
+        public void Dispose()
+        {
+            
+        }
+
+        public void Remove(object key)
+        {
+            cache.Remove(key, out ICacheEntry unused);
+        }
+
+        public bool TryGetValue(object key, out object value)
+        {
+            if (cache.TryGetValue(key, out ICacheEntry existingEntry))
+            {
+                CacheHits++;
+                value = existingEntry.Value;
+                return true;
+            }
+            else
+            {
+                CacheMisses++;
+                value = null;
+                return false;
+            }
+        }
+    }
+
+    public class GitHubClientTests
```

Comment:
Do we want to add some tests to the cache size limit?

Reply:
Added a check that the size is not set to zero upon return.  The underlying caching implementation's behavior is something we can't test from the unit test.